atlas/transactions/admin/fulfill_pack_buyback_offer.cdc

```diff
@@ -0,0 +1,114 @@
+import FungibleToken from 0x{{.FungibleTokenContractAddress}}
+import NonFungibleToken from 0x{{.NonFungibleTokenContractAddress}}
+import DapperUtilityCoin from 0x{{.DapperUtilityCoinContractAddress}}
+import {{.NFTProductName}} from 0x{{.NFTContractAddress}}
+import NFTStorefront from 0x{{.NFTStorefrontV1ContractAddress}}
+
+/// Fulfills a pack buyback offer: dapper pays DUC, user's NFTs are transferred to issuer via NFTStorefront.
+///
+transaction() {
+    // NFT IDs and buyback prices
+    let nftIDs: [UInt64]
+    let prices: [UFix64]
+
+    // User's storefront for listing NFTs
+    let userStorefront: auth(NFTStorefront.CreateListing) &NFTStorefront.Storefront
+
+    // Capability to withdraw user's NFTs for listing
+    let userNFTWithdrawCap: Capability<auth(NonFungibleToken.Withdraw) &{{.NFTProductName}}.Collection>?
+
+    // Capability to receive DUC payment
+    let userDUCReceiverCap: Capability<&{FungibleToken.Receiver}>
+
+    // Dapper's DUC vault and initial balance
+    let dapperVault: auth(FungibleToken.Withdraw) &DapperUtilityCoin.Vault
+    let initialDapperBalance: UFix64
+
+    // Issuer's NFT collection and address
+    let issuerCollection: &NonFungibleToken.Collection
+    let issuerAddress: Address
+
+    prepare(user: auth(Storage, Capabilities) &Account, dapper: auth(BorrowValue) &Account) {
+        // Initialize NFT IDs and buyback prices
+        self.nftIDs = [{{.nftIDs}}]
+        self.prices = [{{.prices}}]
+
+        // Get or create user's storefront
+        if user.storage.borrow<&NFTStorefront.Storefront>(from: NFTStorefront.StorefrontStoragePath) == nil {
+            user.storage.save(<- NFTStorefront.createStorefront(), to: NFTStorefront.StorefrontStoragePath)
+            user.capabilities.publish(
+                user.capabilities.storage.issue<&NFTStorefront.Storefront>(NFTStorefront.StorefrontStoragePath),
+                at: NFTStorefront.StorefrontPublicPath
+            )
+        }
+        self.userStorefront = user.storage.borrow<auth(NFTStorefront.CreateListing) &NFTStorefront.Storefront>(from: NFTStorefront.StorefrontStoragePath)
+            ?? panic("Missing NFTStorefront")
+
+        // Get or create user's NFT withdrawal capability
+        let providerPath = /storage/{{.NFTProductName}}NFTCollectionProviderForNFTStorefront
+        self.userNFTWithdrawCap = user.storage.copy<Capability<auth(NonFungibleToken.Withdraw) &{{.NFTProductName}}.Collection>>(from: providerPath)
```

Comment:
Nit: i know we use this format in existing transactions, but we might want to switch to [cap controllers](https://github.com/onflow/nft-storefront/pull/107/files#diff-be13a65b7079ea7aaf88db2ac71528cd9642a462c597930ab34df8ae45cd67e7R25). Up to you

Reply:
Switched to account controllers and cleaned up redundant account storage - thanks!